dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/IntrinsicsInlining.java

```diff
@@ -0,0 +1,117 @@
+import com.dat3m.dartagnan.expression.INonDet;
+import com.dat3m.dartagnan.expression.type.BooleanType;
+import com.dat3m.dartagnan.expression.type.IntegerType;
+import com.dat3m.dartagnan.expression.type.TypeFactory;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.core.Event;
+import com.dat3m.dartagnan.program.event.functions.DirectFunctionCall;
+import com.dat3m.dartagnan.program.event.functions.DirectValueFunctionCall;
+import com.google.common.primitives.UnsignedInteger;
+import com.google.common.primitives.UnsignedLong;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class IntrinsicsInlining implements ProgramProcessor {
+
+    private int constantId;
+
+    private IntrinsicsInlining() {
+    }
+
+    public static IntrinsicsInlining fromConfig(Configuration ignored) throws InvalidConfigurationException {
+        return new IntrinsicsInlining();
+    }
+
+    @Override
+    public void run(Program program) {
+        constantId = 0;
+        for (Thread thread : program.getThreads()) {
+            run(thread);
+        }
+    }
+
+    private void run(Thread thread) {
+        for (Event event : List.copyOf(thread.getEvents())) {
+            if (!(event instanceof DirectFunctionCall call)) {
+                continue;
+            }
+            assert !call.getCallTarget().hasBody();
+
+            List<Event> replacement = switch (call.getCallTarget().getName()) {
+                case "__VERIFIER_nondet_bool",
+                        "__VERIFIER_nondet_int", "__VERIFIER_nondet_uint", "__VERIFIER_nondet_unsigned_int",
+                        "__VERIFIER_nondet_short", "__VERIFIER_nondet_ushort", "__VERIFIER_nondet_unsigned_short",
+                        "__VERIFIER_nondet_long", "__VERIFIER_nondet_ulong",
+                        "__VERIFIER_nondet_char", "__VERIFIER_nondet_uchar" -> inlineNonDet(call);
+                default -> throw new UnsupportedOperationException(
+                        String.format("Undefined function %s", call.getCallTarget().getName()));
+            };
+
+            call.insertAfter(replacement);
+            call.forceDelete();
+        }
+    }
+
+    private List<Event> inlineNonDet(DirectFunctionCall call) {
+        TypeFactory types = TypeFactory.getInstance();
+        ExpressionFactory expressions = ExpressionFactory.getInstance();
+        assert call instanceof DirectValueFunctionCall;
+        Register register = ((DirectValueFunctionCall) call).getResultRegister();
+        String name = call.getCallTarget().getName();
+        String prefix = name.startsWith("__V") ? "__VERIFIER_nondet_" : "__SMACK_nondet_";
```

Comment:
in line 57, we only call `inlineNonDet` for `__VERIFIER` methods ... why do we need this conditional?